packages/components/src/number-control/index.js

```diff
@@ -93,58 +96,55 @@ export function NumberControl(
 				nextValue = subtract( nextValue, incrementalValue );
 			}
 
-			nextValue = roundClamp( nextValue, min, max, incrementalValue );
-
-			state.value = nextValue;
+			state.value = constrainValue(
+				nextValue,
+				enableShift ? incrementalValue : null
```

Comment:
Minor nit: I think `undefined` is a better fit, as it would equal to not passing the second argument at all

```suggestion
				enableShift ? incrementalValue : undefined
```

Reply:
It's all nullish to me and `constrainValue` 😄 . I would have committed this but the next comment about the same didn't also have the suggestion so I left them both out for now.

What I'd like to point out is that if we proceed with #34817 the `constrainValue` function will not even need the second parameter so this can be made moot. That's how it was in the original version of this PR and part of the reason I just wanted to do it all in one go.

Reply:
Let's leave it as-is, then. It was a minor nit anyway